Automattic/wp-calypso
# Post Editor: reduxify the publish button

This PR introduces a new selector named `getEditorPublishButtonStatus` and uses it to reduxify the publish buttons: the one in editor sidebar:

<img width="316" alt="screen shot 2018-05-22 at 14 30 18" src="https://user-images.githubusercontent.com/664258/40363251-4d936e26-5dcf-11e8-955e-8829e03f4a4c.png">

and also the one in the confirmation sidebar, if enabled:

<img width="273" alt="screen shot 2018-05-22 at 14 43 33" src="https://user-images.githubusercontent.com/664258/40363446-d60951e4-5dcf-11e8-998d-b49c80aad365.png">

The patch is divided into several incremental commits: read the commit messages for details.

**How to test:**
The label of the publish button should reflect the action that is going to happen if you press it: for drafts that are going to be published, it's "Publish". For posts with future publish date it's "Schedule". For already published posts, it's "Update". For users that can't publish and need an approval, it's "Request Review". Verify that all scenarios work.

The button is also enabled only when the post is in a valid state: it needs to have content and a password-protected post must have a valid password. Verify that the button is enabled and disabled as it should be.

When a publish action is in progress, the button should be marked as busy.
